ls1intum/Artemis
# `Development`: Update components to use self closing html tags & add linter rule

### Checklist
#### General



### Motivation and Context

HTML tags for custom components are long and verbose. This leads to a lot of typing, selecting, scrolling, and reading. For a lot of our custom components, we don't even need the closing tags, as they don't have any slots / children.

With HTML linting now enabled, this gives us the opportunity to prefer self-closing html tags in order to make the code less verbose and more readable. And, a diff with -600LOC sounds always good.

### Description

* Added eslint rule which warns when empty closing tags are present: `@angular-eslint/template/prefer-self-closing-tags`
* Ran `eslint --fix` to automatically re-format html w/ closing tags. 
* As `eslint --fix` is run on pre-commit, developers don't really have to do anything, the code will fix itself.

### Steps for Testing

Lint:
* Create HTML tag without content
* eslint warning should appear, suggesting to prefer self-closing tags

Fix:
* Create HTML tag without content
* run `eslint --fix`
* empty html tag should be reformatted to self-closing tag

Nothing to test manually really, it's just `eslint --fix`.
